charts/prometheus/templates/cm.yaml

```diff
@@ -22,11 +22,15 @@ data:
 {{ $root.Values.server.global | toYaml | trimSuffix "\n" | indent 6 }}
 {{- if $root.Values.server.remoteWrite }}
     remote_write:
-{{ $root.Values.server.remoteWrite | toYaml | indent 4 }}
+{{- range $root.Values.server.remoteWrite }}
+{{- tpl . $root | toYaml | cat "-" | nindent 4 }}
+{{- end }}
 {{- end }}
 {{- if $root.Values.server.remoteRead }}
     remote_read:
-{{ $root.Values.server.remoteRead | toYaml | indent 4 }}
+{{- range $root.Values.server.remoteRead }}
+{{- tpl . $root | toYaml | cat "-" | nindent 4 }}
```

Comment:
Good question, looks like this would've failed on any non-string values in the interface :grimacing: Will change things to perform templating on only the `url` field.

Reply:
This issue has now been resolved. For completeness I shall include a few examples of how fields are templated:
1. Normal
    ```yaml
    remoteWrite:
      - url: "https://example.com"
        name: example
    ```
    becomes
    ```yaml
    remote_write:
    - name: example
      url: https://example.com
    ```
2. Templated (release name of "test")
    ```yaml
      - url: "https://{{ Release.Name }}.example.com"
        name: example
    ```
    becomes
    ```yaml
    remote_write:
    - name: example
      url: https://test.example.com
    ```
3. With braces
    ```yaml
    remoteWrite:
    - url: 'https://{{ "{{" }}abnormal{{ "}}" }}.example.com
      name: example
    ```
    becomes
    ```yaml
    remote_write:
      - name: example
        url: https://{{abnormal}}.example.com